### PR TITLE
Table data and queries have been updated

### PR DESCRIPTION
**Pull request by MMhaidari**
# What is added to the DATABASE
- [ ] Add a column species of type string to 

animals table.

- [ ] update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

- [ ] Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.

- [ ] Update the animals table by setting the species column to pokemon for all animals that don't have species already set.

- [ ] Verify that changes were made.

- [ ] Commit the transaction.

- [ ] Inside a transaction delete all records in the animals table, then roll back the transaction.

- [ ] Inside a transaction:
1. Delete all animals born after Jan 1st, 2022.
2. Create a savepoint for the transaction.
3. Update all animals' weight to be their weight multiplied by -1.
4. Rollback to the savepoint
5. Update all animals' weights that are negative to be their weight multiplied by -1.
6. Commit transaction

- [ ] How many animals are there?
- [ ] How many animals have never tried to escape?
- [ ] What is the average weight of animals?

- [ ] Who escapes the most, neutered or not neutered animals

- [ ] What is the minimum and maximum weight of each type of animal

- [ ] What is the average number of escape attempts per animal type of those born between 1990 and 2000

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -5,4 +5,14 @@ VALUES
 	(1,'Agumon', TO_DATE('03/02/2020', 'MM/DD/YYYY'), 0, true, 10.23),
 	(2,'Gabumon', TO_DATE('11/15/2020', 'MM/DD/YYYY'), 2, true, 8),
 	(3,'Pikachu', TO_DATE('01/07/2021', 'MM/DD/YYYY'), 1, false, 15.04),
-	(4,'Devimon', TO_DATE('05/12/2017', 'MM/DD/YYYY'), 5, true, 11);
\ No newline at end of file
+	(4,'Devimon', TO_DATE('05/12/2017', 'MM/DD/YYYY'), 5, true, 11);
+
+INSERT INTO animals (id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES
+		(5,'Charmander', TO_DATE('02/08/2020', 'MM/DD/YYYY'), 0, false, -11),
```
Comment (zieeco) [inline data.sql:12]: - Hey, the data type for the weight is a floating number(decimal). let's ensure we've passed the correct data type to this column `'-11.0'`

File: data.sql [hunk 2]
```diff
@@ -5,4 +5,14 @@ VALUES
...
+		(6,'Plantmon', TO_DATE('11/15/2021', 'MM/DD/YYYY'), 2, true, -5.7),
+		(7,'Squirtle', TO_DATE('04/02/1993', 'MM/DD/YYYY'), 3, false, -12.13),
+		(8,'Angemon', TO_DATE('06/12/2005', 'MM/DD/YYYY'), 1, true, -45),
```
Comment (zieeco) [inline data.sql:15]: - Hey, the data type for the weight is a floating number(decimal). let's ensure we've passed the correct data type to this column `'-45.0'`

File: data.sql [hunk 3]
```diff
@@ -5,4 +5,14 @@ VALUES
...
+		(9,'Boarmon', TO_DATE('06/07/2005', 'MM/DD/YYYY'), 7, true, 20.4),
+		(10,'Blossom', TO_DATE('10/13/1998', 'MM/DD/YYYY'), 3, true, 17),
```
Comment (zieeco) [inline data.sql:17]: - Hey, the data type for the weight is a floating number(decimal). let's ensure we've passed the correct data type to this column `'17.0'`

File: data.sql [hunk 4]
```diff
@@ -5,4 +5,14 @@ VALUES
...
+		(11,'Ditto', TO_DATE('05/14/2022', 'MM/DD/YYYY'), 4, true, -22);
```
Comment (zieeco) [inline data.sql:18]: - Hey, the data type for the weight is a floating number(decimal). let's ensure we've passed the correct data type to this column, and also, this should be a positive integer. `'22.0'` and not `-22.0`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @getssh,



Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice schema file. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions
- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (MMhaidari) [reply data.sql:12]: @zieeco if you pay attention closely we do have integer numbers as well in the same column. I am wondering how to assign different identifiers for the same column. would you please guide me?

File: data.sql (hunk 2 above)
Comment (MMhaidari) [reply data.sql:15]: @zieeco if you pay attention closely we do have integer numbers as well in the same column. I am wondering how to assign different identifiers for the same column. would you please guide me?

File: data.sql (hunk 3 above)
Comment (MMhaidari) [reply data.sql:17]: @zieeco if you pay attention closely we do have integer numbers as well in the same column. I am wondering how to assign different identifiers for the same column. would you please guide me?

File: data.sql (hunk 4 above)
Comment (MMhaidari) [reply data.sql:18]: @zieeco if you pay attention closely we do have integer numbers as well in the same column. I am wondering how to assign different identifiers for the same column. would you please guide me?

**Comment by medaminedev66** [comment]
Hello @MMhaidari👋,

_This is an additional review as a result of the second opinion request._

@zieeco requested changes can be right only on some specific cases. However, based on your cases and the project requirements, it is acceptable to insert an integer number into a decimal column in PostgreSQL. PostgreSQL is designed to handle this scenario intelligently and will appropriately process the integer input without issues.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Comment by MMhaidari** [comment]
# Note: 
- [ ] please read the above second opinion about the project

**Review by OyePriscilla — APPROVED** [review]
Hi @MMhaidari 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Use database transactions ✔️
Modify and delete data in SQL ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**